node/heartbeat/monitor.go

```diff
@@ -236,6 +247,14 @@ func (m *Monitor) ProcessReceivedMessage(message p2p.MessageP2P, fromConnectedPe
 		return err
 	}
 
+	isHardforkTrigger := bytes.Equal(hbRecv.Payload, []byte(hardforkTriggerString))
+	if isHardforkTrigger {
+		errHardforkTrigger := m.hardforkTrigger.TriggerReceived(hbRecv.Pubkey, hbRecv.Payload)
+		if errHardforkTrigger != nil {
+			return errHardforkTrigger
+		}
+	}
+
 	//message is validated, process should be done async, method can return nil
 	go m.addHeartbeatMessageToMap(hbRecv)
```

Comment:
Totally unrelated with this PR, but should this be called on a go routine? The next function is ok, but this one also changes `heartbeatMessages`

Reply:
It is ok to be done in an async manner by at least 2 reasons: 1. the `ProcessReceivedMessage` should return as fast as we can in order to propagate the p2p message to other peers (the return parameter of this function dictates if the message is to be propagated or dropped). Since up to this point, the message proved to be valid. And 2. The further message processing by the current node is not mandatory to be done in sync with the validation flow. This is the model used in all of our interceptors: on the validation flow, check in sync as much as possible to decide if the message is valid or not, node processing such as storing in the data pools done on go routines.

Reply:
Cool, point 2 was exactly what I was having doubts about! Thanks for the clarification